src/main/utils/color.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0
+ * Vesktop, a desktop app aiming to give you a snappier Discord Experience
+ * Copyright (c) 2023 Vendicated and Vencord contributors
+ */
+
+const digitRegex = "([a-zA-Z\\d.%\\*\\(\\)]+)";
+const validFormats = ["rgb", "hsl", "hwb", "lab", "lch", "oklab", "oklch"] as const;
+
+const colorRegexes = [
+    ...validFormats.map(fmt => new RegExp(`(${fmt})\\(${new Array(3).fill(digitRegex).join(" *, *")}\\)`)),
+    ...validFormats.map(fmt => new RegExp(`(${fmt})a\\(${new Array(4).fill(digitRegex).join(" *, *")}\\)`)),
+    ...validFormats.map(fmt => new RegExp(`(${fmt})\\(${new Array(3).fill(digitRegex).join(" +")}\\)`)),
+    ...validFormats.map(
+        fmt => new RegExp(`(${fmt})a?\\(${new Array(3).fill(digitRegex).join(" +")} ?/ ?${digitRegex}\\)`)
+    )
+];
+
+const hexRegex = /^#([\da-fA-F]{6})$/;
+const hexAlphaRegex = /^#([\da-fA-F]{6})[\da-fA-F]{2}$/;
+
+export const alpha = (original: string, alpha: number): string => {
+    for (const regex of colorRegexes) {
+        const match = original.match(regex);
+        if (match) {
+            const [, format, a, b, c] = match;
+            return `${format}a(${a} ${b} ${c} / ${alpha})`;
+        }
+    }
+
+    for (const regex of [hexRegex, hexAlphaRegex]) {
+        const match = original.match(regex);
+        if (match) {
+            const [, str] = match;
+            return `#${str}${Math.round(alpha * 255)
+                .toString(16)
+                .padStart(2, "0")}`;
+        }
+    }
+
+    return original;
+};
```

Comment:
this is still absolute insanity

someone proposed this a while back, why not use it?

```js
function resolveColor(color: string) {
    const span = document.createElement("span")
    span.style.color = color;
    
    document.body.append(span)
    const rgbColor = getComputedStyle(span).color;
    span.remove();
    
    return rgbColor;
}

Reply:
adopted!